### PR TITLE
Support configurable MCS API version

### PR DESCRIPTION
**Pull request by nmittler**
This allows us to support new versions or custom vesions for a given platform.

**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/autoserviceexportcontroller.go [hunk 1]
```diff
@@ -126,9 +124,14 @@ func (c *autoServiceExportController) createServiceExportIfNotPresent(svc *v1.Se
 		},
 	}
 
-	serviceExports := c.client.MulticlusterV1alpha1().ServiceExports(svc.Namespace)
-	_, err := serviceExports.Create(context.TODO(), &serviceExport, metav1.CreateOptions{})
+	// Convert to unstructured.
+	u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&serviceExport)
```
Comment (stevenctl) [inline pilot/pkg/serviceregistry/kube/controller/autoserviceexportcontroller.go:128]: `serviceExport` here is still a `v1alpha1` struct. Can that cause issues?

Comment (nmittler) [reply pilot/pkg/serviceregistry/kube/controller/autoserviceexportcontroller.go:130]: Good catch. Not sure, but added the APIVersion just in case.

**Comment by nmittler** [comment]
@howardjohn @stevenctl This should be ready to go now. PTAL

FYI I had to give the Istio reader write permission for ServiceExport in order for the external control plane tests to pass.

**Review by stevenctl — APPROVED** [review]

File: pkg/test/framework/suite.go [hunk 2]
```diff
@@ -96,8 +96,8 @@ type Suite interface {
 	RequireMaxClusters(maxClusters int) Suite
 	// RequireSingleCluster is a utility method that requires that there be exactly 1 cluster in the environment.
 	RequireSingleCluster() Suite
-	// RequireLocalControlPlane ensures that clusters are using locally-deployed control planes.
-	RequireLocalControlPlane() Suite
+	// RequireMultiPrimary ensures that each cluster is running a control plane.
+	RequireMultiPrimary() Suite
```
Comment (stevenctl) [inline pkg/test/framework/suite.go:100]: +1

File: pkg/kube/register.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	mcs "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	"istio.io/istio/pilot/pkg/features"
+)
+
+var (
+	schemeBuilder = &runtime.SchemeBuilder{}
+
+	// AddToScheme is used to register the MCS CRDs to a runtime.Scheme
+	AddToScheme = schemeBuilder.AddToScheme
```
Comment (howardjohn) [inline pkg/kube/register.go:30]: This should either be scoped under `mcs` folder, or the comment should probably not be MCS specific?

File: pkg/kube/client.go [hunk 4]
```diff
@@ -1087,6 +1051,7 @@ func isEmptyFile(f string) bool {
 var IstioScheme = func() *runtime.Scheme {
 	scheme := runtime.NewScheme()
 	utilruntime.Must(kubescheme.AddToScheme(scheme))
+	utilruntime.Must(AddToScheme(scheme))
```
Comment (howardjohn) [inline pkg/kube/client.go:1054]: should we remove L1061

File: tests/integration/pilot/testdata/mcs-serviceexport-crd.yaml [hunk 5]
```diff
@@ -15,9 +15,9 @@
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
-  name: serviceexports.multicluster.x-k8s.io
+  name: serviceexports.{{ .Group }}
```
Comment (howardjohn) [inline tests/integration/pilot/testdata/mcs-serviceexport-crd.yaml:18]: :nauseated_face:

**Review by howardjohn — APPROVED** [review]

File: tests/integration/pilot/testdata/mcs-serviceexport-crd.yaml (hunk 5 above)
Comment (nmittler) [reply tests/integration/pilot/testdata/mcs-serviceexport-crd.yaml:18]: LOL

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**